java/fory-core/src/main/java/org/apache/fory/Fory.java

```diff
@@ -1572,7 +1573,7 @@ public void resetWrite() {
     metaStringResolver.resetWrite();
     serializationContext.resetWrite();
     bufferCallback = null;
-    depth = 0;
+    depth = -1;
```

Comment:
```suggestion
    depth = 0;
```
I think this should be 0, otherwise the register can still be invoked